Sources/SwiftCompilerPluginMessageHandling/PluginMacroExpansionContext.swift

```diff
@@ -24,6 +24,49 @@ import SwiftSyntax
 import SwiftSyntaxMacros
 #endif
 
+/// Caching parser for PluginMessage.Syntax
+class ParsedSyntaxRegistry {
```

Comment:
Naming nitpick: A registry for me is something that stores long-lived values (possibly even eternal), which is quite the opposite of a cache. Would `ParsedSyntaxCache` be a better name?